src/features/world/scenes/BaseScene.ts

```diff
@@ -403,38 +419,42 @@ export abstract class BaseScene extends Phaser.Scene {
   }
 
   moveOtherPlayers() {
-    for (const sessionId in this.playerEntities) {
-      const room = this.roomService.state.context.rooms[this.roomId];
+    const room = this.roomService.state.context.rooms[this.roomId];
+    if (!room) return;
 
-      if (!room) continue;
+    // Destroy any dereferenced players
+    Object.keys(this.playerEntities).forEach((sessionId) => {
+      if (!room.state.players.get(sessionId)) {
+        this.destroyPlayer(sessionId);
+      }
+    });
 
-      if (sessionId === room.sessionId) continue;
+    // Render current players
+    room?.state.players.forEach((player, sessionId) => {
+      if (sessionId === room.sessionId) return;
```

Comment:
This loop now relies on game state instead of local state. it seems like this state can get a little bit corrupted when restarting scenes. 

It also deletes any stale players from `playerEntities`.